app/CPReflectionService/CPReflectionService.m

```diff
@@ -61,4 +61,14 @@ - (void)allPods:(void (^ _Nonnull)(NSArray<NSString *> * _Nullable pods, NSError
   }];
 }
 
+- (void)updateSpecRepos:(void (^)(NSError * _Nullable))reply
+{
+  [RBObject performBlock:^{
+    [RBObjectFromString(@"Pod::App") update_spec_repos];
```

Comment:
It's not blocking the app's thread, because it's using https://github.com/CocoaPods/CocoaPods-app/blob/master/app/CPReflectionService/RBObject+CocoaPods.m#L195 passing in `NO` for waiting. Not sure about the ruby thread - I can look into spawning off a new ruby thread if that's the case.